src/Symfony/Component/Security/Http/Event/CredentialsValidEvent.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Symfony\Component\Security\Http\Event;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+use Symfony\Contracts\EventDispatcher\Event;
+
+/**
+ * This event is dispatched after authentication has successfully completed.
+ *
+ * At this stage, the authenticator created an authenticated token
+ * and generated an authentication success response. Listeners to
+ * this event can do actions related to successful authentication
+ * (such as migrating the password).
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+class CredentialsValidEvent extends Event
```

Comment:
Same with `CredentialsVerificationFailedEvent` - I'm not sure this really has anything to do with credentials - it's just that authentication was successful.